### PR TITLE
Added code owners based on copyright

### LINKED ISSUE #7: PR #6 shouldn't have been merged so quickly
I was in the midst of writing a review on #6 when it was merged... since the PR was merged, I can't submit my review any more. I don't want to revert it, so sticking my requests here to be addressed in another PR.

Some requests in CONTRIBUTING.md:

 * Line 37: Remove the `cd lalsuite`. Also, could we suggest calling the upstream gwastro instead? This is a matter of personal taste, but I like to always use the names of the organizations/users from which I've forked a repo.
 * Line 54: Add a `get fetch upstream` (or `gwastro`, if you agree with my previous comment) here, before the checkout, to ensure people know that they need to do a fetch first to branch off the most recent changes.

For the CODEOWNERS... is it the case that a code owner *must* approve any PR on that file, or only that they are automatically added as *a* reviewer, with the possibility that a reviewee could request (and get approval from) another reviewer in addition? I'm fine with it if the latter. But if the former, I'd need some convincing... that seems a bit draconian. We'd really need some way of determining who should be added as new code owners, since a file could change substantially over time.

For an initial set of code owners, I'd use the people listed in the copy right notices at the top of all the files. We can have multiple people listed for a file, correct?

### PR DESCRIPTION
**Pull request by duncanmmacleod**
### Short version

This PR adds entries to codeowners for all of the python files in the repo based on the copyright headers in those files.

Fixes #7.

### Long version

@vivienr and I have experimented to the point where we understand the system (which includes our settings for the repo):

- if someone posts a PR that changes code owned by @duncanmmacleod, then I (@duncanmmacleod) will automatically be assigned to review that PR,
- GitHub provides a repo Setting named _**Require review from Code Owners**_, however that would **require** @duncanmmacleod to complete their review before the PR can be merged. This potentially presents a bottleneck if that individual is not available for any reason, so we have opted to **not use that option**,
- without that option @duncanmmacleod will still be automatically assigned to review, however a completed review is not required. Other individuals can be manually assigned, and as long as 1 review is completed, the PR can be merged by a privileged individual.

### Notes

@cdcapano is assigned by default to all files under `/bin/` and `/gwin/` (since he is the copyright owner of the majority of the files), so those few files that don't have a copyright statement are also owned by him.

@micamu is listed as a copyright owner of a few modules, but does not have the right 'status' to be assigned as a reviewer, so her name has not been included. I honestly don't know what github would do in that situation, but there's only one way to find out.

The complete list of named individuals is 

- @cdcapano 
- @cmbiwer
- @duncanmmacleod
- @vivienr

### CODE REVIEW

**Comment by ahnitz** [comment]
Why not just rely on github's ability to choose based on existing contributions to the files you are editing? That ways as someone continues to contribute they will be automatically recommended in the future.

**Comment by cdcapano** [comment]
I'm a bit confused now... what happens if person A is assigned a folder, while person B is assigned one of the files in that folder? Does that mean both A and B are the code owners of that file, or that the file's code owner (B) supersedes the folder's code owner (A)?

**Comment by duncanmmacleod** [comment]
@cdcapano, 

> I'm a bit confused now... what happens if person A is assigned a folder, while person B is assigned one of the files in that folder? Does that mean both A and B are the code owners of that file, or that the file's code owner (B) supersedes the folder's code owner (A)?

A file code owner supersedes the folder code owner.

**Comment by duncanmmacleod** [comment]
@ahnitz,

> Why not just rely on github's ability to choose based on existing contributions to the files you are editing? That ways as someone continues to contribute they will be automatically recommended in the future.

It could end up that this is all we need, but the code owners concept allows us to automatically assign a nominated person regardless of history. There are a number of examples I can think of whereby editing a file does not mean that you understand it (python3 syntax, updating class name, modifying license headers, ...).

**Comment by cdcapano** [comment]
I agree with @ahnitz; I'm not entirely sure this is needed... but I'm willing to give it a try. Want @cmbiwer to weigh in before it's merged though.

**Comment by cmbiwer** [comment]
To me, depends how strict you want this enforced. I prefer to just assign people myself but I don't think this PR really affects that ability as is. As long as I can still add someone else to the review for a PR and I only need 1 approval from either the developer I added myself or the autoperson. Then really I can just carry on as @ahnitz suggests.

**Comment by duncanmmacleod** [comment]
I'm not 100% sold on CODEOWNERS either, for the record, it seemed like a neat idea.

To clarify, if you are automatically assigned by virtue of being a codeowner, you can unassign yourself and assign someone else at will. A PR will still only need 1 approved review to be mergeable.

In practice all this system (as we have configured it) does is auto-assign people based on the contents of the `CODEOWNERS` file, it does not mandate that the code owner actually completes a review. It just absolves some librarian to watch PRs come in and assign a default reviewer manually.

**Review by cdcapano — APPROVED** [review]
Ok, let's try it.

**Review by vivienr — APPROVED** [review]
I'm up for giving it a go.

**Review by cmbiwer — APPROVED** [review]

**Pull request merged**
